homeassistant/components/shelly/utils.py

```diff
@@ -190,7 +191,11 @@ def get_device_wrapper(hass: HomeAssistant, device_id: str):
 async def get_coap_context(hass):
     """Get CoAP context to be used in all Shelly devices."""
     context = aioshelly.COAP()
-    port = hass.data[DOMAIN][CONF_COAP_PORT]
+    port = (
+        hass.data[DOMAIN][CONF_COAP_PORT]
+        if hass.data.get(DOMAIN)
+        else DEFAULT_COAP_PORT
+    )
```

Comment:
We should understand why the config flow test failed, and not fix the code to pass the test.
It may hide a problem were the `coap_context` is created before `hass.data[DOMAIN][CONF_COAP_PORT]` is initialized so that instead of running in the configured port it will run in the default port.

Reply:
Back to original implementation and fixed tests